openmetadata-ui/src/main/resources/ui/src/hooks/useClipBoard.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ *  Copyright 2023 Collate.
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+import { toNumber } from 'lodash';
+import { useCallback, useEffect, useState } from 'react';
+
+/**
+ * React hook to copy text to clipboard
+ * @param value the text to copy
+ * @param timeout delay (in ms) to switch back to initial state once copied.
+ * @param callBack execute when content is copied to clipboard
+ */
+export const useClipboard = (
+  value: string,
+  timeout = 1500,
+  callBack?: () => void
+) => {
+  // local state
+  const [hasCopied, setHasCopied] = useState(false);
+  const [valueState, setValueState] = useState(value);
+
+  // handlers
+  const handleCopy = useCallback(async () => {
+    try {
+      await navigator.clipboard.writeText(valueState);
```

Comment:
Can you verify the compatibility of this feature in other browsers?
https://caniuse.com/?search=clipboard

Reply:
@chirag-madlani , Yes, I did and it has good support for all major browsers.